### PR TITLE
Closes #4980 Refactor Kinsta thirdparty class

### LINKED ISSUE #4980: RUCSS - Kinsta's cache isn't cleared after successfully creating the used CSS
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ `3.11.0.5`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

After the successful creation of the used CSS we are purging the URLs cache:
https://github.com/wp-media/wp-rocket/blob/b1d16b7bd8ddfb0a998330fea09cd3fd687fb92f/inc/Engine/Cache/PurgeActionsSubscriber.php#L53

When a site is hosted on Kinsta the cache purging will not take place automatically, and used CSS will be applied only during the next cache purging.

**To Reproduce**

1. On a Kinsta website enable Remove Unused CSS.
2. Make sure that used CSS has been created for a URL.
3. Check the URL. The used CSS will not have been applied.
4. Clear the cache and check again. Used CSS should be there.

**Expected behavior**

On Kinsta, a URL's cache should be cleared after the used CSS has been created.

**Additional context**

In Kinsta's compatibility file, we should hook `rocket_clean_kinsta_post_cache()` to:
* `rocket_rucss_complete_job_status`
* `rocket_rucss_after_clearing_usedcss`

**Ticket:** https://secure.helpscout.net/conversation/1862503536/339693/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description
This PR is here to fix two problems:
- Refactor Kinsta into new architecture
- Fix a problem where cache wasn't empty after RUCSS.

For the first issue we deleted old code and create a thirdparty class in the new architecture.
For the second issue we hooked `rocket_clean_kinsta_post_cache` method to two events:
-   `rocket_rucss_complete_job_status`
-   `rocket_rucss_after_clearing_usedcss` 

Fixes #4980 

## Type of change

- [ ] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

Yes changed the function to `rocket_clean_kinsta_cache_url` instead of `rocket_clean_kinsta_post_cache` on  the second part.

## How Has This Been Tested?

- [ ] Automated tests

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CrochetFeve0251** [comment]
@Tabrisrp I would need you help to do integration tests on this one cause I got no idea how to do it and didn't found anything similar in others Thirdparty classes.

**Comment by remyperona** [comment]
@CrochetFeve0251 For the integration tests, you can do as it is done for other hostings, adding an entry in the `integration/bootstrap.php` file, and a new test in `composer.json`

**Comment by Tabrisrp** [comment]
@CrochetFeve0251 For the integration tests, you can do as it is done for other hostings, adding an entry in the `integration/bootstrap.php` file, and a new test in `composer.json`

File: inc/ThirdParty/Hostings/Kinsta.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
+<?php
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/Kinsta.php:1]: In this file:
- You can use the `WP_Rocket\ThirdParty\ReturnTypesTrait` methods to return  `false` or an empty array where necessary without creating new methods
- You can remove all `@author` references
- You can remove all `rocket_` prefixes from the methods names

File: inc/ThirdParty/Hostings/Kinsta.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	 *
+	 * @param string $root WP Rocket root cache path.
+	 * @param string $lang Current language.
+	 * @return void
+	 */
+	public function rocket_clean_kinsta_cache_home( $root = '', $lang = '' ) {
+		$url = get_rocket_i18n_home_url( $lang );
+		$url = trailingslashit( $url ) . 'kinsta-clear-cache/';
+
+		wp_remote_get(
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/Kinsta.php]:
```suggestion
		wp_safe_remote_get(
```

File: inc/ThirdParty/Hostings/Kinsta.php [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	 * @since 3.0.4
+	 * @author Remy Perona
+	 *
+	 * @param string $url URL to purge.
+	 * @return void
+	 */
+	public function rocket_clean_kinsta_cache_url( $url ) {
+		$url = trailingslashit( $url ) . 'kinsta-clear-cache/';
+
+		wp_remote_get(
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/Kinsta.php]:
```suggestion
		wp_safe_remote_get(
```

File: inc/ThirdParty/Hostings/Kinsta.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	public function rocket_cache_mandatory_cookies() {
+		return [];
+	}
+
+	/**
+	 * Display notice when we are on Kinsta but the plugin is not present.
+	 *
+	 * @return void
+	 */
+	public function display_error_notice() {
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/Kinsta.php:142]: This notice should be displayed only if the global `$kinsta_cache` is not set

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -164,16 +165,17 @@
 			"@test-integration-pdfembeddersecure",
 			"@test-integration-multisite",
 			"@test-integration-cloudways",
-		  	"@test-integration-wpengine",
+		  "@test-integration-wpengine",
```
Comment (engahmeds3ed) [inline composer.json]: plz adjust those spacings

File: composer.json [hunk 6]
```diff
@@ -164,16 +165,17 @@
...
 			"@test-integration-dreampress",
 			"@test-integration-godaddy",
 			"@test-integration-revolutionslider",
-		    "@test-integration-litespeed",
-		    "@test-integration-wordfence",
-		    "@test-integration-convertplug"
+		  "@test-integration-litespeed",
+		  "@test-integration-wordfence",
+		  "@test-integration-kinsta",
+		  "@test-integration-convertplug"
```
Comment (engahmeds3ed) [inline composer.json:176]: here too

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]
@Tabrisrp do we need to deprecate old functions in the deleted kinsta file?

**Comment by remyperona** [comment]
Yes indeed we should

**Review by vmanthos — APPROVED** [review]
I tested:
* the original issue described in #4980
* the refactored class to make sure our compatibility works the way it used to before the latest changes.

Both are working as expected. 👍

**Pull request merged**